spec/requests/private_controller_spec.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe PrivateController, type: :controller do
```

Comment:
```suggestion
RSpec.describe PrivateController, type: :request do
```

Reply:
Good insight from this. I modified all `controller` specs to use `request` which is more favored than the old `controller`. In the process I had to add `config.hosts` as described by this [thread](https://stackoverflow.com/a/60573259/3736373). 🙏 

Please help me check if I understand `request` tests correctly.